### ISSUE TITLE
trouble with "brew install cutest --with-matlab"  A compilation error occurred

### ISSUE THREAD

**Issue by optcutestinstall** [optimizers/homebrew-cutest#49]
I am trying to install "CUTEst" basis with homebrew-cutest on MacOSX.
After inputting "brew install cutest --with-matlab", I got this:

==> Installing cutest from optimizers/cutest
==> Installing dependencies for optimizers/cutest/cutest: optimizers/cutest/sifdecode
==> Installing optimizers/cutest/cutest dependency: optimizers/cutest/sifde
==> Downloading https://github.com/ralna/SIFDecode/archive/v2.0.1.tar.gz
Already downloaded: /Users/***/Library/Caches/Homebrew/downloads/d3df1afe978b450e8a3164c0e71f16292693f88b01d1e9aeb0c49af917112de7--SIFDecode-2.0.1.tar.gz
==> ./install_sifdecode < sifdecode.input
<cmd-i Last 15 lines from /Users/***/Library/Logs/Homebrew/sifdecode/01.install_sifdecode:
collect2: error: ld returned 1 exit status>
[ OK ]
 sifdecode: tools packages (double precision version) compiled successfully
 Compiling random         		[ OK ]
 Compiling check_derivs   		[ OK ]
 Compiling check_derivs_main		[ OK ]
 sifdecode: check packages (double precision version) compiled successfully
 Compiling slct           	<cmd-i	ld: unexpected token: !tapi-tbd-v3 file '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd' for architecture x86_64
collect2: error: ld returned 1 exit status
make[1]: *** [/private/tmp/sifdecode-20190706-4815-73p6c9/SIFDecode-2.0.1/objects/mac64.osx.gfo/double/slct.o] Error 1
make: *** [all_double] Error 2>

 ERROR: A compilation error occurred.

Does anyone know how to solve this problem?

**Comment by dpo** [comment]
Could you post the output of `brew gist-logs sifdecode`?

**Comment by optcutestinstall** [comment]
$ brew gist-logs sifdecode
You can create a new personal access token:
 https://github.com/settings/tokens/new?scopes=gist,public_repo&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.bash_profile

**Comment by optcutestinstall** [comment]
brew gist-logs sifdecode
You can create a new personal access token:
  https://github.com/settings/tokens/new?scopes=gist,public_repo&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.bash_profile

GitHub User: optcutestinstall
Password: 

Error: GitHub Requires authentication:The GitHub credentials in the macOS keychain may be invalid.
Clear them with:
  printf "protocol=https\nhost=github.com\n" | git credential-osxkeychain erase
Or create a personal access token:
  https://github.com/settings/tokens/new?scopes=gist,public_repo&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.bash_profile

**Comment by dpo** [comment]
Did you create a personal token as instructed?

**Comment by optcutestinstall** [comment]
Yes. 
It generates a long string.




At 2019-07-06 19:42:48, "Dominique" <notifications@github.com> wrote:


Did you create a personal token as instructed?

**Comment by dpo** [comment]
You have to add it to your `~/.bash_profile` as instructed:
> `echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.bash_profile`
and open a new terminal.

**Comment by optcutestinstall** [comment]
Thanks. I run the command again:

$ brew tap optimizers/cutest

$ brew install cutest --with-matlab

==>Installing cutest from optimizers/cutest

==>Installing dependencies for optimizers/cutest/cutest: optimizers/cutest/sifdecode

==>Installing optimizers/cutest/cutest dependency: optimizers/cutest/sifde

==>Downloading https://github.com/ralna/SIFDecode/archive/v2.0.1.tar.gz

Already downloaded: /Users/**/Library/Caches/Homebrew/downloads/d3df1afe978b450e8a3164c0e71f16292693f88b01d1e9aeb0c49af917112de7--SIFDecode-2.0.1.tar.gz

==>./install_sifdecode < sifdecode.input

Last 15 lines from /Users/**/Library/Logs/Homebrew/sifdecode/01.install_sifdecode:

collect2: error: ld returned 1 exit status

[ OK ]

 sifdecode: tools packages (double precision version) compiled successfully

 Compiling random         [ OK ]

 Compiling check_derivs   [ OK ]

 Compiling check_derivs_main[ OK ]

 sifdecode: check packages (double precision version) compiled successfully

 Compiling slct           ld: unexpected token: !tapi-tbd-v3 file '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd' for architecture x86_64

collect2: error: ld returned 1 exit status

make[1]: *** [/private/tmp/sifdecode-20190706-16296-556y9u/SIFDecode-2.0.1/objects/mac64.osx.gfo/double/slct.o] Error 1

make: *** [all_double] Error 2




 ERROR: A compilation error occurred.







If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):

  https://github.com/optimizers/homebrew-cutest/issues




/usr/local/Homebrew/Library/Homebrew/utils/github.rb:253:in `raise_api_error': GitHub Bad credentials:HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check: (GitHub::AuthenticationFailedError)

  https://github.com/settings/tokens

from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:212:in `open_api'

from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:357:in `search'

from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:265:in `search_issues'

from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:278:in `issues_for_formula'

from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:374:in `fetch_issues'

from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:370:in `issues'

from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:424:in `dump'

from /usr/local/Homebrew/Library/Homebrew/brew.rb:141:in `rescue in <main>'

from /usr/local/Homebrew/Library/Homebrew/brew.rb:38:in `<main>'







At 2019-07-06 20:00:48, "Dominique" <notifications@github.com> wrote:


You have to add it to your ~/.bash_profile as instructed:

echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.bash_profile
and open a new terminal.

**Comment by dpo** [comment]
Make sure that you copied your token correctly (to state the obvious, "your_token_here" should be replaced with the token string in your `~/.bash_profile`).

**Comment by optcutestinstall** [comment]
I replace  "your_token_here" by the token string, and input 
brew gist-logs sifdecode again. 
It returns :

https://gist.github.com/cdf539b8e888faf3cac08d84224dfce9







At 2019-07-06 20:17:55, "Dominique" <notifications@github.com> wrote:


Make sure that you copied your token correctly (to state the obvious, "your_token_here" should be replaced with the token string in your ~/.bash_profile).

**Comment by dpo** [comment]
Thank you. Now I see the error message. Could you try: `sudo xcode-select -switch /Library/Developer/CommandLineTools` and `brew install sifdecode` again? If it doesn't work, please copy again the output of `brew gist-logs sifdecode`.

**Comment by optcutestinstall** [comment]
https://gist.github.com/16eb7772d8b19146409901f39503fb53







At 2019-07-06 20:37:19, "Dominique" <notifications@github.com> wrote:


Thank you. Now I see the error message. Could you try: sudo xcode-select -switch /Library/Developer/CommandLineTools and brew install sifdecode again? If it doesn't work, please copy again the output of brew gist-logs sifdecode.

**Comment by dpo** [comment]
You don't really need Xcode to build and use CUTEst. The command-line tools (CLT) are sufficient. Do you need Xcode for anything else? If not, try to uninstall Xcode. I use the same version of the CLT as you do and all works well for me.

If you absolutely need Xcode, there is a solution at the bottom of this page: https://wiki.classe.cornell.edu/ACC/ACL/CodeTroubleshooting.

**Comment by optcutestinstall** [comment]
Thank you. Let me try again.




At 2019-07-06 20:46:39, "Dominique" <notifications@github.com> wrote:


You don't really need Xcode to build and use CUTEst. The command-line tools (CLT) are sufficient. Do you need Xcode for anything else? If not, try to uninstall Xcode. I use the same version of the CLT as you do and all works well for me.

If you absolutely need Xcode, there is a solution at the bottom of this page: https://wiki.classe.cornell.edu/ACC/ACL/CodeTroubleshooting.

**Comment by dpo** [comment]
Did it work?

**Comment by optcutestinstall** [comment]
I found that I installed MacPorts on my computer. After uninstalling MacPorts, the installation of cutest succeeded






At 2019-07-06 22:21:54, "Dominique" <notifications@github.com> wrote:


Did it work?

**Comment by dpo** [comment]
Great!
